firebase/firestore/_utils.py

```diff
@@ -29,8 +29,14 @@ def _from_datastore(data):
 
 	for key, val in data_to_restructure.items():
 
-		if val.get('mapValue'):
-			data_to_restructure[key] = _from_datastore(val['mapValue'])
+		if isinstance(val.get('mapValue'), dict):
+			dic = {}
+
+			if val['mapValue'].get('fields'):
+				for field_key, field_val in val['mapValue']['fields'].items():
+					dic[field_key] = _decode_datastore(field_val)
```

Comment:
this would cause another bug, because the array field as a value for a key can't be decoded by `_decode_datastore` function. A proposal change for the whole thing (line 32-39) would be

```python

if isinstance(val.get('mapValue'), dict):

	if val['mapValue'].get('fields', False):
		data_to_restructure[key] = _from_datastore(val['mapValue'])
	else:
		data_to_restructure[key] = {}

```

Reply:
Good catch. I fixed it according to your proposal in [8decc9d](https://github.com/AsifArmanRahman/firebase-rest-api/pull/13/commits/8decc9d5d0ece00fc66634a8da1c9c1a74fec814)